### PR TITLE
Fixed DatasetVersionDao queries to return input and output facets

### PR DESCRIPTION
**Pull request by dominiquetipton**
Fixed DatasetVersionDao queries to return input and output facets.

Fixed bug in DatasetDao findAll query and slightly reformatted other queries.

Related to Issue #1360 and PR #1477.

### CODE REVIEW

**Review by collado-mike — COMMENTED** [review]
Thanks for fixing this 👍🏽

Can we add something to the `DatasetIntegrationTest` to validate the input/output facets are working as expected? E.g., in one of the tests, after the dataset is created post a LineageEvent like
```java
LineageEvent.builder()
                .eventType("COMPLETE")
                .run(new LineageEvent.Run(firstRunId.toString(), RunFacet.builder().build()))
                .job(LineageEvent.Job.builder().name(JOB_NAME).namespace(NAMESPACE).build())
                .eventTime(Instant.now().atZone(TIMEZONE))
                .inputs(new ArrayList<>())
                .outputs(Collections.singletonList(new Dataset(
        NAMESPACE_NAME,
        DB_TABLE_NAME,
        LineageTestUtils.newDatasetFacet(
            ImmutableMap.of("outputFacetKey", "outputFacetValue"),
            new SchemaField("firstname", "string", "the first name")))))
                .build()
```

Fetching the DatasetVersion should then return the input and output facets.

**Comment by dominiquetipton** [comment]
Sure thing @collado-mike 👍

File: api/src/test/java/marquez/OpenLineageIntegrationTest.java [hunk 1]
```diff
@@ -112,12 +119,34 @@ private void validateDatasetFacets(JsonNode json) {
...
+
+    DatasetVersion latestDatasetVersion = datasetVersions.get(0);
+    if (!expectedFacets.isMissingNode()) {
+      assertThat(latestDatasetVersion.getNamespace()).isEqualTo(namespace);
+      assertThat(latestDatasetVersion.getName()).isEqualTo(output);
+      final JsonNode facetsForDatasetVersion =
+          Utils.getMapper().convertValue(latestDatasetVersion.getFacets(), JsonNode.class);
+      assertThat(facetsForDatasetVersion).isEqualTo(expectedFacets);
+    } else {
+      assertThat(latestDatasetVersion.getFacets()).isEmpty();
```
Comment (collado-mike) [inline api/src/test/java/marquez/OpenLineageIntegrationTest.java:149]: This is great!

**Review by collado-mike — APPROVED** [review]
Awesome! Thanks for getting this in so quickly 👍🏽

File: api/src/test/java/marquez/OpenLineageIntegrationTest.java (hunk 1 above)
Comment (dominiquetipton) [reply api/src/test/java/marquez/OpenLineageIntegrationTest.java:149]: Thanks Mike!

**Review by wslulciuc — APPROVED** [review]
LGTM 👍 💯

**Pull request merged**